cmd/kubeadm/app/util/kubeconfig/kubeconfig_test.go

```diff
@@ -345,3 +346,198 @@ func TestHasCredentials(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterFromKubeConfig(t *testing.T) {
+	tests := []struct {
+		name      string
+		config    *clientcmdapi.Config
+		expected  string
+		expected1 *clientcmdapi.Cluster
+	}{
+		{
+			name: "cluster is empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "kubernetes",
+			},
+			expected:  "",
+			expected1: nil,
+		},
+		{
+			name: "cluster and currentContext are not empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"foo": {AuthInfo: "foo", Cluster: "foo"},
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expected: "foo",
+			expected1: &clientcmdapi.Cluster{
+				Server: "http://foo:8080",
+			},
+		},
+		{
+			name: "cluster is not empty and currentContext is not in contexts",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expected:  "",
+			expected1: nil,
+		},
+	}
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			got, got1 := GetClusterFromKubeConfig(rt.config)
```

Comment:
doesn't GetClusterFromKubeConfig return one value of type *clientcmdapi.Cluster?

https://github.com/my-git9/kubernetes/blob/0424c7c74d926b4fe3193059e003e9056b429d28/cmd/kubeadm/app/util/kubeconfig/kubeconfig.go#L107

Reply:
Thanks for your review, maybe my local code needs to be updated